contributed_definitions/nyaml/NXscan_control.yaml

```diff
@@ -0,0 +1,1066 @@
+  traj_SCAN(NXobject):
+    doc: |
+      To define the trajectory scan, use this group.
+    scan_speed(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        Define the scan speed through the trajectory points.
+    forward_speed(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        The field defines the scan speed in the forward directions through the
+        trajectory points.
+    backward_speed(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        The field defines the scan speed in the backward directions (backtracking)
+        through the trajectory points.
+    channel_NAME_N:
+      doc: |
+        Name of the channel that records the scan data for the given dimension. The ending annotation 'N' is
+        to represent the name of the dimensions.
+        
+        Rename the field, according to the name of the channel and dimension (e.g. piezo_scanner_x).
+    number_of_trajectory_points(NX_NUMBER):
+      doc: |
+        The number of trajectory points in the entire scan.
+    trajectory_points(NX_NUMBER):
+      doc: |
+        The trajectory points are the N-dimensional vectors describing all the scan points sequentially.
+        
+        The second rank dataset should contain total number of trajectory points (nTraj) and
+        a full coordinate (nD) of each trajectory point.
+      dimensions:
+        rank: 2
+        dim: [[1, nTraj], [2, nD]]
+    scan_points_N(NX_NUMBER):
+      doc: |
+        Define the total number of scan points between two trajectory points.
+        
+        Rename the field, according to the index of the second trajectory points.
+        For example, scan_points_1 is the number of points between the starting(zero) and first trajectory points.
+    stepping_N(NX_NUMBER):
+      doc: |
+        The number of steps the probe jumps over the scan steps or points along the trajectory line from between two trajectory points.
+        
+        Rename the field, according to the index of the second trajectory points.
+        For example, scan_points_1 is the number of points between the starting(zero) and first trajectory points.
+      \@frog_leap_scan:
+        type: NX_BOOLEAN
+        doc: |
+          If the scan probe jumps over a number of scan steps or points (more than one)
+          to perform each scan. By default, it is False.
+    step_size(NX_NUMBER):
+      unit: NX_ANY
+      doc: |
+        The length of each step along the entire trajectory line.
+    continuous(NX_BOOLEAN):
+      doc: |
+        If the scan probe moves continuously over the scan points or steps between two
+        trajectory points, use True. The default value is True.
+    oscillating_N(NX_BOOLEAN):
+      doc: |
+        If the scan probe oscillates over the scan point, use True.
+        The default value is False.
+    oscillation_frequency(NX_NUMBER):
+      unit: NX_FREQUENCY
+      doc: |
+        The number of oscillations on each scanning point per second.
+    SCAN_DATA(NXdata):
```

Comment:
This is problematic because by default, the underscore in not renameable. 
You should either
1) set `nameType=any` (i.e., make the whole thing renameable) or
2) use something like `SCANDATA` without the underscore

Reply:
At a discussion with sandor we found that `nameType` is not going to be implemented right now. That should come along with all the app defs, base classes from all domains. So, currently we can skip it.

Reply:
But this is independent. The way that you have it now, it will fail the validation. Why not just use `SCANDATA`?

Reply:
I can write only `DATA`, but it does not carry any intrinsic meaning, e.g., which data it refers.

So, if `SCAN_DATA` is not valid, is it because `_` is not part of the variadic name, i.g. it can be renamed to `x_y` not to `xy`. Right?

Reply:
Yes, exactly. The `_` part is not renameable, you will always have `somepart_anotherpart`, but never ``somepartanotherpart` (unless you set `nameType=any` for the whole thing).

Why not use `SCAN_data` with `nameType=partial` then, if you want to retain the `_data` part anymore?

Reply:
noted tobe implementd!

Reply:
Renamed to `SCAN_data`.